gef.py

```diff
@@ -689,10 +689,33 @@ def size(self) -> int:
             raise AttributeError
         return self.page_end - self.page_start
 
+    def search_for_realpath(self) -> str:
+        """This function is a workaround for gdb bug #23764
+
+        path might be wrong for remote sessions, so try a simple search for files
+        that aren't found at the path indicated, which should be canonical.
+        """
+
+        remote_path = pathlib.Path(self.path)
+        # First, try the canonical path in the remote session root.
+        candidate1 = gef.session.remote.root / remote_path.relative_to(remote_path.anchor)
+        if candidate1.is_file():
+            return str(candidate1)
+
+        # On some systems, /lib(64) might be a symlink to /usr/lib(64), so try removing
+        # the /usr prefix.
+        if remote_path.is_relative_to("/usr"):
+            candidate = gef.session.remote.root / remote_path.relative_to("/usr")
+            if candidate.is_file():
```

Comment:
Wouldn't this check always be true?

Reply:
No, I don't think so.  This check modifies the path in a way that might be true for one type of remote system, but it's just a guess.  It's not guaranteed to match a path after modifying it.